src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs

```diff
@@ -36,6 +43,14 @@ internal static void VerifyArguments(Diagnostic diagnostic, Func<Diagnostic, boo
             {
                 throw new ArgumentException(string.Format(CodeAnalysisResources.UnsupportedDiagnosticReported, diagnostic.Id), nameof(diagnostic));
             }
+
+            if (string.IsNullOrEmpty(diagnostic.Id) || !diagnostic.Id.All(UnicodeCharacterUtilities.IsIdentifierPartCharacter))
```

Comment:
@gafter Thanks for the clarification, I have addressed all your feedback. Kindly review one more time :)